### PR TITLE
use accept header to determine best image format

### PR DESCRIPTION
**Pull request by Jared1204**

### CODE REVIEW

File: src/common/image-optimisation.ts [hunk 1]
```diff
@@ -33,14 +34,16 @@ export const optimiseImage = async (
     };
   }
 
-  const desiredImageFormat = canAcceptAvif ? ImageTypes.avif : ImageTypes.webp;
+  logger.info({
+    message: `converting image to format: ${format}`,
+  });
 
-  pipe.toFormat(desiredImageFormat, {
+  pipe.toFormat(format as unknown as AvailableFormatInfo, {
```
Comment (Liam-Johnston) [inline src/common/image-optimisation.ts:41]: Can we have the type declaration in the parameters match what is required here please

Comment (Jared1204) [reply src/common/image-optimisation.ts:41]: i cant for the life of me get it to convert neatly to their type for this

File: src/controllers/image-request.ts [hunk 2]
```diff
@@ -1,30 +1,62 @@
...
+    } else headerValues.push(value);
   });
 
-  return false;
+  if (headerValues.includes('*/*') || headerValues.includes('image/avif')) {
+    logger.info({
+      message: `best format is: ${ImageTypes.avif}`,
+    });
+    return ImageTypes.avif;
+  } else if (headerValues.includes('image/webp')) {
```
Comment (Liam-Johnston) [inline src/controllers/image-request.ts:37]: No ned for the else here and all the statements below

File: src/controllers/image-request.ts [hunk 3]
```diff
@@ -1,30 +1,62 @@
...
-    return false;
+    logger.info({
+      message: 'no accept header',
+    });
+    return ImageTypes.webp;
   }
 
+  // get all the header values out incase there is multiple 'accept' headers
+  const headerValues: string[] = [];
   acceptHeader.forEach(({ value }) => {
```
Comment (Liam-Johnston) [inline src/controllers/image-request.ts:23]: would a `.reduce` make this clearer?

File: src/controllers/image-request.ts [hunk 4]
```diff
@@ -1,30 +1,62 @@
...
   if (acceptHeader === undefined) {
-    return false;
+    logger.info({
+      message: 'no accept header',
+    });
+    return ImageTypes.webp;
   }
 
+  // get all the header values out incase there is multiple 'accept' headers
+  const headerValues: string[] = [];
```
Comment (Liam-Johnston) [inline src/controllers/image-request.ts:22]: would a `.reduce` make this clearer/ more concise?

**Review by Liam-Johnston — CHANGES_REQUESTED** [review]
Just a few minor things here

File: src/common/image-optimisation.ts (hunk 1 above)
Comment (Liam-Johnston) [reply src/common/image-optimisation.ts:41]: yeah all good 👍

File: src/controllers/image-request.ts (hunk 4 above)
Comment (Jared1204) [reply src/controllers/image-request.ts:22]: not really, we still need to split out each header value incase there is multiple accept headers passed and also catering for each one being able to have multiple accepts in it

Comment (Liam-Johnston) [reply src/controllers/image-request.ts:22]: `.includes` will behave the same in a single string or in an array

Comment (Liam-Johnston) [reply src/controllers/image-request.ts:22]: In this implementation

Comment (Liam-Johnston) [reply src/controllers/image-request.ts:22]: alternatively if you want to keep it as an array the outcome can be achieved by doing this:
```
headerValues = acceptHeader.reduce(
  (previousValue, { value }) => previousValue + value,
  ''
).split(',')
```

**Review by Liam-Johnston — APPROVED** [review]

**Pull request merged**